libcst/codemod/visitors/_apply_type_annotations.py

```diff
@@ -0,0 +1,558 @@
+# Copyright (c) 2016-present, Facebook, Inc.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree
+
+
+from typing import Dict, List, NamedTuple, Optional, Sequence, Set, Tuple, Union
+
+import libcst as cst
+from libcst.codemod._context import CodemodContext
+from libcst.codemod._visitor import ContextAwareTransformer
+from libcst.codemod.visitors._gather_imports import GatherImportsVisitor
+
+
+def _get_name_or_attribute_as_string(
+    name_or_attribute: cst.BaseExpression,
+) -> Optional[str]:
+    if isinstance(name_or_attribute, cst.Name):
+        result = name_or_attribute.value
+    elif isinstance(name_or_attribute, cst.Attribute):
+        value = _get_name_or_attribute_as_string(name_or_attribute.value)
+        if value is None:
+            return None
+        attribute_name = name_or_attribute.attr.value
+        result = f"{value}.{attribute_name}"
+    else:
+        result = None
+    return result
+
+
+def get_import_alias_names(import_aliases: Sequence[cst.ImportAlias]) -> Set[str]:
+    import_names = set()
+    for imported_name in import_aliases:
+        asname = imported_name.asname
+        if asname:
+            import_names.add(_get_name_or_attribute_as_string(asname.name))
+        else:
+            import_names.add(_get_name_or_attribute_as_string(imported_name.name))
+    return import_names
+
+
+def get_import_names(imports: Sequence[Union[cst.Import, cst.ImportFrom]]) -> Set[str]:
+    import_names = set()
+    for _import in imports:
+        if isinstance(_import, cst.Import):
+            import_names.update(get_import_alias_names(_import.names))
+        else:
+            names = _import.names
+            if not isinstance(names, cst.ImportStar):
+                import_names.update(get_import_alias_names(names))
+    return import_names
+
+
+class FunctionAnnotation(NamedTuple):
```

Comment:
@jimmylai I think the rest uses dataclasses, is there a reason to stick with namedtuple here?